simba/schedule.py

```diff
@@ -967,3 +1147,54 @@ def generate_event_list_from_prices(
         if start_events > start_simulation or stop < stop_simulation:
             logging.info(f"{gc_name} price csv does not cover simulation time")
     return events
+
+
+def get_charge_delta_soc(charge_curves: dict, vt: str, ct: str, max_power: float,
+                         duration_min: float, start_soc: float) -> float:
+    charge_curve = charge_curves[vt][ct][max_power]
+    d_soc = optimizer_util.get_delta_soc(charge_curve, start_soc, duration_min=duration_min)
+    return d_soc
+
+
+def soc_at_departure_time(v_id_deps: tuple, departure_time, vehicle_data, stations, charge_curves,
+                          args):
+    """ Get the possible soc of the vehicle of a rotation at a specified departure_time
+
+    :param v_id_deps: vehicle_id and depot name
```

Comment:
highly suggest to separate vehicle_id and depot name into individual arguments. See suggestion for line 371